lib/Reader/readerir.cpp

```diff
@@ -6441,6 +6445,12 @@ bool GenIR::interlockedIntrinsicBinOp(IRNode *Arg1, IRNode *Arg2,
   }
 
   if (Op != AtomicRMWInst::BinOp::BAD_BINOP) {
+    ASSERT(Arg1->getType()->isPointerTy());
```

Comment:
We've been migrating towards using assert instead of ASSERT so new code should use assert.